Sources/NIOHPACK/DynamicHeaderTable.swift

```diff
@@ -0,0 +1,166 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// Implements the dynamic part of the HPACK header table, as defined in
+/// [RFC 7541 § 2.3](https://httpwg.org/specs/rfc7541.html#dynamic.table).
+struct DynamicHeaderTable {
+    public static let defaultSize = 4096
+    
+    /// The actual table, with items looked up by index.
+    private var storage: HeaderTableStorage
+    
+    /// The length of the contents of the table.
+    var length: Int {
+        return self.storage.length
+    }
+    
+    /// The size to which the dynamic table may currently grow.
+    var allowedLength: Int {
+        get {
+            return self.storage.maxSize
+        }
+        set {
+            self.storage.setTableSize(to: newValue)
+        }
+    }
+    
+    var maximumTableLength: Int {
```

Comment:
I think this and `allowedLength` need more documentation to distinguish them from each other.